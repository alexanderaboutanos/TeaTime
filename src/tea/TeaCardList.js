/** @format */

import { Row } from "react-bootstrap";
import TeaCard from "../tea/TeaCard";

const TeaCardList = ({ teas, origin, btn1, btn2 }) => {
  console.log(teas);

  return (
    <Row xs={1} md={4} className="TeaCardList g-4">
      {teas.map((tea) => (
        <TeaCard
          key={tea.id}
          id={tea.id}
          title={tea.title}
          brand={tea.brand}
          img_url={tea.img_url || tea.image}
          description={tea.description}
          btn1={btn1}
          btn2={btn2}
          origin={origin}
        />
      ))}
    </Row>
  );
};

export default TeaCardList;
