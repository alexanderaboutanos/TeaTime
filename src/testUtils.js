/** @format */

import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser1",
  first_name: "Test",
  last_name: "User",
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

const fakeTeas = [
  {
    id: 740259,
    title:
      "Tea Sampler - Fruit Tea - Antioxidants Tea - Rooibos Tea - 100% Natural - Decaffeinated - Blueberry Tea - Peach Tea - Raspberry Tea - Loose Leaf Tea",
    brand: "Chinese Tea Culture",
    description:
      "Tea Sampler - Fruit Tea - Antioxidants Tea - Rooibos Tea - 100% Natural - Decaffeinated - Blueberry Tea - Peach Tea - Raspberry Tea - Loose Leaf Tea",
    img_url: "https://spoonacular.com/productImages/740259-312x231.jpeg",
  },
  {
    id: 1539729,
    title:
      "Tea Forte Lotus Relaxing Teas Presentation Box Tea Sampler Gift Set, 20 Assorted Variety Handcrafted Pyramid Tea Infuser Bags, Black Tea, Green Tea, Oolong Tea, White Tea, Herbal Tea",
    brand: "Tea Forte",
    description:
      "LOTUS COLLECTION, a curated assortment of five relaxing hot teas for mind, body & spirit - a unique and unprecedented gourmet collection of highest-grade, small-lot, certified organic teas: Vanilla Pear, Orange Jasmine, Mountain Oolong, Darjeeling Quince, Lemon Lavender DIVERSE, SUBTLE FLAVORS from only the finest whole tea leaves in the world, blended with aromatic fruits, flowers and herbs DELIGHTFUL TEA GIFT SET offers a variety of premium gourmet teas, a most welcomed hostess gift or gift for tea lovers PRESENTATION BOX tea sampler has 20 assorted pyramid tea bag infusers, handcrafted to allow the delicate leaves to luxuriantly unfurl in hot water, producing a deliciously aromatic, flavorful cup WRAPPED IN ELEGANT, embossed pattern gift box with a satin ribbon, open the lid to reveal a tea tasting menu with blend descriptions",
    img_url: "https://spoonacular.com/productImages/1539729-312x231.jpeg",
  },
];

export { UserProvider, fakeTeas };
