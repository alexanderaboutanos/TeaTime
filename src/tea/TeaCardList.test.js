/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { fakeTeas } from "../testUtils";
import TeaCardList from "./TeaCardList";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <TeaCardList teas={fakeTeas} />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <TeaCardList teas={fakeTeas} w />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
