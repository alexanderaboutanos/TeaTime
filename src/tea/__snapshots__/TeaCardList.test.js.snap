// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TeaCardList g-4 row row-cols-md-4 row-cols-1"
  >
    <div
      class="col"
    >
      <div
        class="h-100 card"
        id="740259"
      >
        <img
          class="card-img-top"
          src="https://spoonacular.com/productImages/740259-312x231.jpeg"
          style="cursor: pointer;"
        />
        <div
          class="bg-light card-body"
          style="cursor: pointer;"
        >
          <div
            class="text-truncate card-title h5"
          >
            Tea Sampler - Fruit Tea - Antioxidants Tea - Rooibos Tea - 100% Natural - Decaffeinated - Blueberry Tea - Peach Tea - Raspberry Tea - Loose Leaf Tea
          </div>
          <p
            class="card-text"
          >
            Chinese Tea Culture
          </p>
          <p
            class="crop-text-2 card-text"
          >
            Tea Sampler - Fruit Tea - Antioxidants Tea - Rooibos Tea - 100% Natural - Decaffeinated - Blueberry Tea - Peach Tea - Raspberry Tea - Loose Leaf Tea
          </p>
        </div>
        <div
          class="card-footer"
        >
          <button
            class="btn btn-primary btn-sm"
            style="margin: 3px;"
            type="button"
          />
          <button
            class="btn btn-primary btn-sm"
            type="button"
          />
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="h-100 card"
        id="1539729"
      >
        <img
          class="card-img-top"
          src="https://spoonacular.com/productImages/1539729-312x231.jpeg"
          style="cursor: pointer;"
        />
        <div
          class="bg-light card-body"
          style="cursor: pointer;"
        >
          <div
            class="text-truncate card-title h5"
          >
            Tea Forte Lotus Relaxing Teas Presentation Box Tea Sampler Gift Set, 20 Assorted Variety Handcrafted Pyramid Tea Infuser Bags, Black Tea, Green Tea, Oolong Tea, White Tea, Herbal Tea
          </div>
          <p
            class="card-text"
          >
            Tea Forte
          </p>
          <p
            class="crop-text-2 card-text"
          >
            LOTUS COLLECTION, a curated assortment of five relaxing hot teas for mind, body & spirit - a unique and unprecedented gourmet collection of highest-grade, small-lot, certified organic teas: Vanilla Pear, Orange Jasmine, Mountain Oolong, Darjeeling Quince, Lemon Lavender DIVERSE, SUBTLE FLAVORS from only the finest whole tea leaves in the world, blended with aromatic fruits, flowers and herbs DELIGHTFUL TEA GIFT SET offers a variety of premium gourmet teas, a most welcomed hostess gift or gift for tea lovers PRESENTATION BOX tea sampler has 20 assorted pyramid tea bag infusers, handcrafted to allow the delicate leaves to luxuriantly unfurl in hot water, producing a deliciously aromatic, flavorful cup WRAPPED IN ELEGANT, embossed pattern gift box with a satin ribbon, open the lid to reveal a tea tasting menu with blend descriptions
          </p>
        </div>
        <div
          class="card-footer"
        >
          <button
            class="btn btn-primary btn-sm"
            style="margin: 3px;"
            type="button"
          />
          <button
            class="btn btn-primary btn-sm"
            type="button"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
